src/colorizer/CanvasUIOverlay.ts

```diff
@@ -71,12 +61,16 @@ type RenderInfo = {
   render: () => void;
 };
 
+const EMPTY_RENDER_INFO: RenderInfo = { sizePx: new Vector2(0, 0), render: () => {} };
+
 /**
- * A canvas used for drawing UI overlays over another screen region. (intended for use
- * with `ColorizeCanvas`.)
+ * Extends the ColorizeCanvas class by overlaying and compositing additional
+ * dynamic elements (like a scale bar, timestamp, etc.) on top of the
+ * base rendered image.
  */
-export default class CanvasOverlay {
-  public readonly canvas: OffscreenCanvas;
+export default class CanvasOverlay extends ColorizeCanvas {
```

Comment:
We don't use ColorizeCanvas separately anymore, yeah. I do like that all of the colorizing-related logic is inside the ColorizeCanvas class...

I could make `ColorizeCanvas` a publicly-accessible instance inside of `CanvasOverlay`  (the way the overlay + canvas used to be). The con is that there is a lot of internal state (like the color ramp min/max, color, categorical palette, dataset, current feature key, etc.) that should also be accessible to `CanvasOverlay`, and would need to be made public. (Unfortunately no such thing as a `friend` class in TS/JS.)

Reply:
> Also: do we ever use a ColorizeCanvas alone, anymore? It kinda seems to me that these could now be merged into one class that just manages both canvases... but I do recognize that it might be a very big class. I only say this because it's hard to understand the notion that a CanvasOverlay "is a" ColorizeCanvas. I guess I have a preference for aggregation over inheritance in general.

One thing I did a while back but deleted was make an `IControllableCanvas` interface. I've brought it back here so that classes outside of `CanvasWrapper` aren't tightly coupled with `ColorizeCanvas`. Can you let me know if this is better, or do you think it still warrants refactoring `CanvasWithOverlay` to not inherit from `ColorizeCanvas`?